lib/hawkular/inventory/inventory_api_v4.rb

```diff
@@ -57,14 +57,26 @@ def children_resources(parent_id)
     # @return [Array<Resource>] List of resources
     def root_resources
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get('/resources?root=true')['results'].map { |r| Resource.new(r) }
+      resources root: "true"
+    end
+
+    # List resources
+    # @param [Hash] filter options to filter the resource list
+    # @option filter :root If truthy, only get root resources
+    # @option filter :feedId Filter by feed id
+    # @option filter :typeId Filter by type id
+    def resources(filter = {})
+      # FIXME: pagination => lazy-loaded list with ruby?
+      filter[:root] = !! filter[:root] if filter.key? :root
```

Comment:
`filter.fetch(:root) { false }`

Reply:
I need to cast it to boolean also, any suggestion?

Reply:
`!!`, if you really really need a boolean